datafusion/core/tests/parquet/schema_adapter.rs

```diff
@@ -370,3 +378,321 @@ async fn test_custom_schema_adapter_and_custom_expression_adapter() {
     ];
     assert_batches_eq!(expected, &batches);
 }
+
+// ----------------------------------------------------------------------
+// Tests migrated from schema_adaptation/schema_adapter_integration_tests.rs
+// ----------------------------------------------------------------------
+
+/// A schema adapter factory that transforms column names to uppercase
+#[derive(Debug, PartialEq)]
+struct UppercaseAdapterFactory {}
+
+impl SchemaAdapterFactory for UppercaseAdapterFactory {
```

Comment:
Moving the tests here sort of implies they are only related to parquet -- don't we apply schema adapter to other formats too?

However, since all the tests use parquet this seems like a good place to put them